src/azure-cli-core/azure/cli/core/commands/client_factory.py

```diff
@@ -83,6 +84,12 @@ def configure_common_settings(cli_ctx, client):
     client.config.enable_http_logger = True
 
     client.config.add_user_agent(UA_AGENT)
+
+    try:
+        client.config.add_user_agent(os.environ[ENV_AZ_INSTALLER])
+    except KeyError:
+        pass
```

Comment:
The same logic is also needed by `az rest`: https://github.com/Azure/azure-cli/blob/31a9724478c67751ae0bee6cc0d9b75b763df17c/src/azure-cli-core/azure/cli/core/util.py#L556-L558

Maybe we can extract the header handling logic to `src/azure-cli-core/azure/cli/core/util.py`?

Reply:
#12734 is merged. We can merge from dev and put the logic in `get_az_user_agent`.